reactor-core/src/main/java/reactor/core/publisher/FluxDematerialize.java

```diff
@@ -86,32 +68,29 @@ public void onSubscribe(Subscription s) {
 				this.s = s;
 
 				actual.onSubscribe(this);
-
-				s.request(1);
 			}
 		}
 
 		@Override
 		public void onNext(Signal<T> t) {
 			if (done) {
+				//TODO interpret the Signal and drop differently?
 				Operators.onNextDropped(t, actual.currentContext());
 				return;
 			}
+
 			if (t.isOnComplete()) {
 				s.cancel();
 				onComplete();
 			}
-			else if (t.isOnError()) {
+			else if (t.isOnError() && t.getThrowable() != null) {
```

Comment:
1) I'm curious how we can get `onError` without the error?
2) Since we add these guards, I wonder whether we should handle the `else` case now (currently it is unhandled and may block the processing)

Reply:
ah yeah, leftover from trying to make the IDE happy with the test and nullability. Seems IntelliJ cannot infer that `hasError()` implies non-null `getThrowable()`...